sdk/metric/sdk.go

```diff
@@ -474,29 +424,11 @@ func (r *record) RecordOne(ctx context.Context, number core.Number) {
 }
 
 func (r *record) Unbind() {
-	for {
-		collected := atomic.LoadInt64(&r.collectedEpoch)
-		modified := atomic.LoadInt64(&r.modifiedEpoch)
-
-		updated := collected + 1
-
-		if modified == updated {
-			// No change
-			break
-		}
-		if !atomic.CompareAndSwapInt64(&r.modifiedEpoch, modified, updated) {
-			continue
-		}
-
-		if modified < collected {
-			// This record could have been reclaimed.
-			r.labels.meter.saveFromReclaim(r)
-		}
-
-		break
-	}
-
-	_ = atomic.AddInt64(&r.refcount, -1)
+	// Record was modified, inform the Collect() that things need to be collected.
+	// TODO: Reconsider if we should marked as modified when an Update happens and
+	// collect only when updates happened even for Bounds.
+	atomic.StoreInt64(&r.modified, 1)
```

Comment:
Yeah, I think that `r.modified` could be incremented every time the update happens (so it would be then named `r.modificationCount` or something), and zeroed on collection with CAS.

Reply:
Actually I don't think I care to count the number of updates, only if an update happens so just a store with 1 is enough. Will leave the TODO for the moment to not change too many things in this PR.

Reply:
The count of updates isn't important in itself - this count would only be used when zeroing it with CAS to catch a case when there was a new update between checkpointing the record and zeroing the count of updates, so we could run checkpoint again. Not sure if this is important - if the update happens, we probably can wait until next collection time. But I was wondering if it is possible that Collect checkpoints, then update happens, Collect zeroes the counter with atomic store, and then the update won't be checkpointed until another update happens or the handle is unbound.

Reply:
But yeah, that can wait for another PR, for sure.

Reply:
Will follow after this with more small fixes.

Reply:
FWIW, I feel unconcerned about allowing records that have not been modified to be output during collection. The rest of the SDK can tell when there are no values.

Reply:
(And the tests already cover this case. There's a race between inserting a new record and actually updating the aggregator with the new data. Because of this case, I'm not sure we need to track whether the record was modified.

Reply:
I would like still to clarify what is the expected behavior, will deal with that in the next PR and discuss there.